source/node/examples/node-quickstart.ts

```diff
@@ -0,0 +1,108 @@
+import * as Realm from "realm";
+import { ObjectId } from "bson";
+
+// Update this with your App ID
+const app = new Realm.App({ id: "<Your App ID>" });
+
+type Task = {
+  _id: ObjectId;
+  _partition?: string;
+  name: string;
+  status: string;
+};
+const TaskSchema = {
+  name: "Task",
+  properties: {
+    _id: "objectId",
+    _partition: "string?",
+    name: "string",
+    status: "string",
+  },
+  primaryKey: "_id",
+};
+
+async function run() {
+  const credentials = Realm.Credentials.anonymous();
+  await app.logIn(credentials);
+  console.log(`Logged in anonymously with user id: ${app.currentUser?.id}`);
+
+  const realm: Realm = await Realm.open({
+    schema: [TaskSchema],
+    sync: {
+      user: app.currentUser as Realm.User,
+      partitionValue: "quickstart",
+    },
+  });
+
+  // Get all Tasks in the realm
+  const tasks: Realm.Collection<Task> = realm.objects<Task>("Task");
+
+  // Add a listener that fires whenever one or more Tasks are inserted, modified, or deleted.
+  tasks.addListener(taskListener);
+
+  // Add a couple of Tasks in a single, atomic transaction
+  // Realm automatically sets the _partition property based on the partitionValue used to open the realm
+  realm.write(() => {
+    const task1: Task = realm.create("Task", {
+      _id: new ObjectId(),
+      name: "go grocery shopping",
+      status: "Open",
+    });
+
+    const task2: Task = realm.create("Task", {
```

Comment:
Explicit type is understandable for a demo, but we already know it's a task from above. Prefer type inference?
```suggestion
    const task2 = realm.create("Task", {
```

Reply:
Yeah my thought here was demo -> more explicit = more better, but why not show both explicit and inferred typing?